sabakan/score.go

```diff
@@ -5,45 +5,60 @@ import (
 )
 
 const (
-	baseScore      = 10
-	unhealthyScore = -3 * baseScore
+	// maxCountPerRack should be more than max machine num per rack + 1
+	maxCountPerRack = 100
+	// heathyScore is addted when the machine status is healthy.
+	heathyScore = 1000
 )
 
 func scoreByDays(days int) int {
 	var score int
 	if days > 250 {
-		score += baseScore
+		score++
 	}
 	if days > 500 {
-		score += baseScore
+		score++
 	}
 	if days > 1000 {
-		score += baseScore
+		score++
 	}
 	if days < -250 {
-		score -= baseScore
+		score--
 	}
 	if days < -500 {
-		score -= baseScore
+		score--
 	}
 	if days < -1000 {
-		score -= baseScore
+		score--
 	}
 	return score
 }
 
 func scoreMachine(m *Machine, rackCount map[int]int, ts time.Time) int {
+	rackScore := maxCountPerRack - rackCount[m.Spec.Rack]
+
 	days := int(m.Spec.RetireDate.Sub(ts).Hours() / 24)
-	score := scoreByDays(days)
-	k := rackCount[m.Spec.Rack]
-	if k > baseScore {
-		k = baseScore
-	}
-	score -= k
+	daysScore := scoreByDays(days)
+
+	return rackScore*10 + daysScore
+}
 
+func scoreMachineWithHealthStatus(m *Machine, rackCount map[int]int, ts time.Time) int {
+	score := scoreMachine(m, rackCount, ts)
 	if m.Status.State != StateHealthy {
-		score += unhealthyScore
+		return score
 	}
+	return heathyScore + score
+}
 
-	return score
+func filterHealthyMachinesByRole(ms []*Machine, role string) []*Machine {
+	var filtered []*Machine
+	for _, m := range ms {
+		if m.Status.State != StateHealthy || (role != "" && m.Spec.Role != role) {
+			continue
+		}
```

Comment:
I'd like to suggest a bit refactor for better readability.

```suggestion
		if m.Status.State != StateHealthy {
			continue
		}
		if role != "" && m.Spec.Role != role {
			continue
		}
```